starfish/core/spots/_detect_pixels/combine_adjacent_features.py

```diff
@@ -267,8 +267,20 @@ def _single_spot_attributes(
                 'x': int(spot_property.centroid[1])
             }
 
-        # we're back to 3d or fake-3d here
-        target_index = decoded_image[spot_attrs['z'], spot_attrs['y'], spot_attrs['x']]
+        # define the target index as the most repeated value in the bounding box of the spot.
+        #  it appears there is no simpler way to do this with a regionprops object
+        bbox = spot_property.bbox
+        if len(bbox) == 6:
+            # 3d bbox
+            target_candidates = np.ndarray.flatten(
+                decoded_image[bbox[0]:bbox[3], bbox[1]:bbox[4], bbox[2]:bbox[5]])
+        else:
+            # flatten and remove zeros
+            target_candidates = np.ndarray.flatten(
+                decoded_image[0, bbox[0]:bbox[2], bbox[1]:bbox[3]])
+        # get the most repeated nonzero value
+        non_zero_target_candidates = target_candidates[target_candidates != 0]
+        target_index = np.argmax(np.bincount(non_zero_target_candidates))
```

Comment:
This requires quite a lot of introspection to understand how it's working. It'd be good to drop a comment in here. Why remove zeros?